framework/utils/frontUtils.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Repository of utils for non-dominated and Pareto frontier methods
+  Created  Feb 18, 2020
+  @authors: Diego Mandelli
+"""
+# External Imports
+import numpy as np
+# Internal Imports
+
+
+
+def nonDominatedFrontier(data, returnMask):
+  """
+    This method is designed to identify the set of non-dominated points (nEfficientPoints)
+    @ In, data, np.array, data matrix (nPoints, nCosts) containing the data point
```

Comment:
Is the data limited to two dimensions?

Reply:
Yes, we are dealing here with point sets in an N dimensional space where the number of dimensions is equal to nCosts.
As of now the Pareto frontier can deal only with 2 dimensional data (nCosts=2).
This core engine extend the capabilities to an arbitrary number of dimensions.